### PR TITLE
add releasing.md

### PR DESCRIPTION
**Pull request by tuminoid**
Describe the release process to release Metal3, ie. CAPM3 with latest
IPAM and BMO. This process then points to newly updates releasing.md
in each of the components.

### CODE REVIEW

File: processes/releasing.md
Comment (dtantsur) [inline processes/releasing.md:27]: nit: ironic-image

File: processes/releasing.md
Comment (dtantsur) [inline processes/releasing.md:37]: nit: MariaDB

File: processes/releasing.md
Comment (dtantsur) [inline processes/releasing.md:56]: Maybe provide an example or at least the line(s) to change?

File: processes/releasing.md
Comment (dtantsur) [inline processes/releasing.md:79]: BMO does not contain these two, maybe change the wording?

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:56]: That's the plan. This PR is marked WIP as I'm documenting the required changes as were doing them in 1.4 release cycle.

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:79]: This is WIP. The wording will be changed quite a bit, especially everything around BMO.

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:27]: Fixed

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:37]: Fixed

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:62]: Done.

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:135]: Done. Let me know if the wording is OK, and if not, please propose a better wording.

**Comment by tuminoid** [comment]
/cc @dtantsur @elfosardo @furkatgofurov7 @kashifest 

PTAL. We released CAPM3 1.4 and documented all the steps needed to take. Some of them were added here, some were added to CAPM3, IPAM and BMO documents, and I also linked them to this document.

**Comment by tuminoid** [comment]
Lol tests that take seconds have been running for 10 hours. I’ll rebase or something tomorrow.

File: processes/releasing.md
Comment (lentzi90) [inline processes/releasing.md:69]:
```suggestion
[Dev-env PR](https://github.com/metal3-io/metal3-dev-env/pull/1222)
```

File: processes/releasing.md
Comment (lentzi90) [inline processes/releasing.md:64]: I'm not understanding what this means. Is it something about the combinations that should be tested?

**Review by lentzi90 — COMMENTED** [review]
Nice job documenting all of this!

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:64]: Yes, and the dev-env example PR is there to help understand what needs to be changed. I can try reword it a bit.

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:69]: Whoops, fixed.

File: processes/releasing.md
Comment (dtantsur) [reply processes/releasing.md:131]: I'm not sure what you're intending to say. Maybe "BMO makes strong assumption about the exact configuration of Ironic. However, [ironic-image](https://github.com/metal3-io/ironic-image) is currently branchless."

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md]: Done, simplified it. Is it any better now?

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md]: Thanks. Used your wording and cut down a lot of the details, they were unnecessary here.

**Review by lentzi90 — COMMENTED** [review]
/lgtm

File: processes/releasing.md
Comment (lentzi90) [reply processes/releasing.md]: Thanks much clearer!

**Review by kashifest — DISMISSED** [review]

File: processes/releasing.md
Comment (furkatgofurov7) [inline processes/releasing.md:19]:
```suggestion
If a new minor or major release was published, we need to make changes in Metal3 dev-env
```

File: processes/releasing.md
Comment (furkatgofurov7) [inline processes/releasing.md:50]: Something like, "Prior art" (commonly used in CAPI for referencing example previous PR) may suit here and in all occurrences below (Example PR) better:
```suggestion
[Prior art](https://gerrit.nordix.org/c/infra/cicd/+/17709) that adds support
```

File: processes/releasing.md
Comment (furkatgofurov7) [inline processes/releasing.md:69]:
```suggestion
Dev-tools synchronises `metal3-io` changes to the `Nordix` fork for the EST
```

File: processes/releasing.md
Comment (furkatgofurov7) [inline processes/releasing.md:60]: for consistency:
```suggestion
### Configure Metal3 dev-env to support new branches
```

File: processes/releasing.md
Comment (furkatgofurov7) [inline processes/releasing.md:59]: Once project infra PR is merged (assuming CI jobs are ready, release branches exits and so on), the new jobs that are made as required in the infra PR (i.e in above PR, it is `test-ubuntu-integration-release-1-4`) has to be also configured in corresponding GitHub repo branch protection rules, otherwise it is useless and GitHub won't be seeing them as mandatory. Let's take a CAPM3 as an example and how to do it:

- trigger a new job (`test-ubuntu-integration-release-1-4`) in the PR (could be test PR) pointing to that branch (release-1.4) at least once so that it can be configured in later steps
- someone with admin rights in the org, has to go to https://github.com/metal3-io/cluster-api-provider-metal3/settings/branches > Click `Edit` on release-1.4 branch > Paste the new test job in search bar in `Status checks that are required.` and select it from drop down > Click on "Save changes"

**Review by furkatgofurov7 — COMMENTED** [review]
Mostly nits, great job @tuminoid

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:50]: Fixed

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:69]: Fixed

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:60]: Fixed

File: processes/releasing.md
Comment (tuminoid) [reply processes/releasing.md:59]: This is documented per repository in their release documentation, see linked PRs.

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Comment by tuminoid** [comment]
I need approval for this, @dtantsur @zaneb @furkatgofurov7

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Review by furkatgofurov7 — APPROVED** [review]
/lgtm cancel
/approve

**Pull request merged**
